komorebi/settings.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import os
+from enum import Enum
+
+from gi.repository import Gio, GLib
+
+import komorebi.consts
+
+key_file_group : str = 'KomorebiProperties'
+
+# Settings
+wallpaper_name : str = 'foggy_sunny_mountain'
+time_twenty_four : bool = True
+autostart : bool = True
+show_desktop_icons : bool = True
+enable_video_wallpapers : bool = True
+mute_playback : bool = False
+pause_playback : bool = True
+
+class keys(Enum):
+	WALLPAPER_NAME = 'WallpaperName'
+	TIME_TWENTY_FOUR = 'TimeTwentyFour'
+	AUTOSTART = 'Autostart'
+	SHOW_DESKTOP = 'ShowDesktopIcons'
+	ENABLE_VIDEO = 'EnableVideoWallpapers'
+	MUTE_PLAYBACK = 'MutePlayback'
+	PAUSE_PLAYBACK = 'PausePlayback'
+
+_config_key_file : GLib.KeyFile = None
+_config_file : Gio.File = None
+
+def _optional(value, default):
+	return value if value is not None else default
+
+def load_settings() -> None:
+	global wallpaper_name
+	global time_twenty_four
+	global autostart
+	global show_desktop_icons
+	global enable_video_wallpapers
+	global mute_playback
+	global pause_playback
+	global _config_key_file
+	global _config_file
+
+	logging.info("Loading configuration...")
+	logging.debug(f'Config dir is "{get_config_dir()}"')
+
+	config_path = os.path.join(get_config_dir(), 'komorebi.prop')
+	_config_file = Gio.File.new_for_path(config_path)
+	_config_key_file = GLib.KeyFile()
+
+	# If the file doesn't exist, then perform first setup
+	if not _config_file.query_exists():
+		bootstrap_config_path()
+		if not _config_file.query_exists():
+			logging.info("No configuration file found. Creating a new one...")
+			save_configuration_file()
+			return
+	
+	logging.info("Reading config file...")
+
+	_config_key_file.load_from_file(config_path, GLib.KeyFileFlags.NONE)
+
+	if not _config_key_file.has_group(key_file_group):
+		logging.warning('Invalid configuration file found, Fixing...')
+		save_configuration_file()
+
+	wallpaper_name = _optional(_config_key_file.get_string(key_file_group, keys.WALLPAPER_NAME.value), 'foggy_sunny_mountain')
+	time_twenty_four = _optional(_config_key_file.get_boolean(key_file_group, keys.TIME_TWENTY_FOUR.value), True)
+	autostart = _optional(_config_key_file.get_boolean(key_file_group, keys.AUTOSTART.value), False)
+	show_desktop_icons = _optional(_config_key_file.get_boolean(key_file_group, keys.SHOW_DESKTOP.value), True)
+	enable_video_wallpapers = _optional(_config_key_file.get_boolean(key_file_group, keys.ENABLE_VIDEO.value), True)
+	mute_playback = _optional(_config_key_file.get_boolean(key_file_group, keys.MUTE_PLAYBACK.value), False)
+	pause_playback = _optional(_config_key_file.get_boolean(key_file_group, keys.PAUSE_PLAYBACK.value), True)
```

Comment:
Maybe put this in a dictionary and use a loop here? This is just an improvement, don't feel pressured to do it in this PR, just something to keep in mind.